### PR TITLE
Reference documentation

### PR DESCRIPTION
**Pull request by a-szotyori**
Added KDoc comments and updated visibility modifiers.

### Ticket 
[PIA-448](https://ginis.atlassian.net/browse/PIA-448)

### CODE REVIEW

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt [hunk 1]
```diff
@@ -27,13 +27,55 @@ private const val EXTRA_IN_EXTRACTIONS = "EXTRA_IN_EXTRACTIONS"
 
 private const val EXTRA_IN_COMPOUND_EXTRACTIONS = "EXTRA_IN_COMPOUND_EXTRACTIONS"
 
+/**
+ * When you use the Screen API, the `DigitalInvoiceActivity` displays the line items extracted from an invoice document and their total
+ * price. The user can deselect line items which should not be payed and also edit the quantity, price or description of each line item.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt]: "payed" -> "paid for"

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt [hunk 2]
```diff
@@ -27,13 +27,55 @@ private const val EXTRA_IN_EXTRACTIONS = "EXTRA_IN_EXTRACTIONS"
...
+ * The total price is always updated to include only the selected line items.
+ *
+ * The returned extractions in the [CameraActivity.EXTRA_OUT_EXTRACTIONS] and [CameraActivity.EXTRA_OUT_COMPOUND_EXTRACTIONS] are updated to
+ * include the user's modifications:
+ * - "amountToPay" was updated to contain the sum of the selected line items' prices,
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt]: is updated

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt [hunk 3]
```diff
@@ -27,13 +27,55 @@ private const val EXTRA_IN_EXTRACTIONS = "EXTRA_IN_EXTRACTIONS"
...
+ * - the line items were updated according to the user's modifications.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt]: are updated

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt [hunk 4]
```diff
@@ -27,13 +27,55 @@ private const val EXTRA_IN_EXTRACTIONS = "EXTRA_IN_EXTRACTIONS"
...
+ * - line item extractions have been enabled for your client id
+ * - the analysis result contains line item extractions
+ *
+ * ### Customizing the Digital Invoice Screen
+ *
+ * Customizing the look of the Digital Invoice Screen is done via overriding of app resources.
+ *
+ * The following items are customizable:
+ * - TODO
+ * - TODO
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt]: Still todo?

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt [hunk 5]
```diff
@@ -26,6 +26,31 @@ private const val ARGS_COMPOUND_EXTRACTIONS = "GV_ARGS_COMPOUND_EXTRACTIONS"
 private const val TAG_RETURN_REASON_DIALOG = "TAG_RETURN_REASON_DIALOG"
 private const val TAG_WHAT_IS_THIS_DIALOG = "TAG_WHAT_IS_THIS_DIALOG"
 
+/**
+ * When you use the Component API the `DigitalInvoiceFragment` displays the line items extracted from an invoice document and their total
+ * price. The user can deselect line items which should not be payed and also edit the quantity, price or description of each line item. The
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt]: "payed" -> "paid for"

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt [hunk 6]
```diff
@@ -26,6 +26,31 @@ private const val ARGS_COMPOUND_EXTRACTIONS = "GV_ARGS_COMPOUND_EXTRACTIONS"
...
+ * total price is always updated to include only the selected line items.
+ *
+ * The returned extractions in the [DigitalInvoiceFragmentListener.onPayInvoice()] are updated to include the user's midifications:
+ * - "amountToPay" was updated to contain the sum of the selected line items' prices,
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt]: is updated

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt [hunk 7]
```diff
@@ -26,6 +26,31 @@ private const val ARGS_COMPOUND_EXTRACTIONS = "GV_ARGS_COMPOUND_EXTRACTIONS"
...
+ * - the line items were updated according to the user's modifications.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceFragment.kt]: are updated

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/LineItem.kt [hunk 8]
```diff
@@ -28,20 +33,40 @@ class LineItem(
         val rawGrossPrice: String
 ) : Parcelable {
 
+    /**
+     * The parsed gross price.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/LineItem.kt]: This is a unit price? Maybe would be clearer to state it.

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt [hunk 9]
```diff
@@ -6,28 +6,70 @@ import android.os.Bundle
...
+ * When you use the Screen API, the `LineItemDetailsActivity` displays a line item to be edited by the user. The user can modify the
+ * following:
+ * - deselect the line item,
+ * - edit the line item description,
+ * - edit the quantity,
+ * - edit the price.
+ *
+ * The extractions returned in the [CameraActivity.EXTRA_OUT_COMPOUND_EXTRACTIONS] are updated to include the user's modifications.
+ *
+ * The `LineItemDetailsActivity` is started by the [DigitalInvoiceActivity] when a user taps on a
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt]: the user

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt [hunk 10]
```diff
@@ -6,28 +6,70 @@ import android.os.Bundle
...
+ * line item to edit it.
+ *
+ * ### Customizing the Line Item Details Screen
+ *
+ * Customizing the look of the Line Item Details Screen is done via overriding of app resources.
+ *
+ * The following items are customizable:
+ * - TODO
+ * - TODO
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt]: TODO here as well

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsFragmentListener.kt [hunk 11]
```diff
@@ -7,7 +7,18 @@ import net.gini.android.vision.digitalinvoice.SelectableLineItem
...
  * Copyright (c) 2019 Gini GmbH.
  */
+
+/**
+ * Interface used by the [LineItemDetailsFragment] to dispatch events to the hosting Activity.
+ */
 interface LineItemDetailsFragmentListener {
 
+    /**
+     * Called when the user pressed the save button.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsFragmentListener.kt]: "presses". Also in other places.

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsFragmentListener.kt [hunk 12]
```diff
@@ -7,7 +7,18 @@ import net.gini.android.vision.digitalinvoice.SelectableLineItem
...
+     *
+     * The selectable line item was updated to contain the user's modifications.
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsFragmentListener.kt]: is updated

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/package.md [hunk 13]
```diff
@@ -0,0 +1,15 @@
+# Package net.gini.android.vision.digitalinvoice.details
+
+Contains the Activity and Fragments used for the return assistant's Line Item Details Screen.
+
+## Screen API
+
+The [net.gini.android.vision.digitalinvoice.details.LineItemDetailsActivity] is launched by the
+[net.gini.android.vision.digitalinvoice.DigitalInvoiceActivity] when a user tapped on a line item to edit it. It displays the line item and
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/package.md]: the user taps

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/package.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
+# Package net.gini.android.vision.digitalinvoice
+
+Contains the Activity and Fragments used for the return assistant's Digital Invoice Screen.
+
+## Screen API
+
+The [net.gini.android.vision.digitalinvoice.DigitalInvoiceActivity] is launched by the [net.gini.android.vision.analysis.AnalysisActivity] when line
+item extractions are available. It displays the line items extracted from an invoice document and their total price. The user can deselect line
+items which should not be payed and also edit the quantity, price or description of each line item. The total price is always updated to
```
Comment (maciejtrybilo) [inline ginivision/src/main/java/net/gini/android/vision/digitalinvoice/package.md]: "paid for"

File: screenapiexample/src/main/java/net/gini/android/vision/screen/ExtractionsActivity.kt [hunk 15]
```diff
@@ -51,6 +51,10 @@ class ExtractionsActivity : AppCompatActivity() {
         setUpRecyclerView()
     }
 
+    private fun deleteMe() {
```
Comment (maciejtrybilo) [inline screenapiexample/src/main/java/net/gini/android/vision/screen/ExtractionsActivity.kt]: It wants to be deleted :)

File: screenapiexample/src/main/java/net/gini/android/vision/screen/MainActivity.java [hunk 16]
```diff
@@ -373,6 +373,10 @@ private void bindViews() {
         mGiniApiTypeSpinner = findViewById(R.id.gini_api_type_spinner);
     }
 
+    private void deleteMe() {
```
Comment (maciejtrybilo) [inline screenapiexample/src/main/java/net/gini/android/vision/screen/MainActivity.java]: This one too.

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt (hunk 4 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/digitalinvoice/DigitalInvoiceActivity.kt]: Customization stuff I want to do in this ticket https://ginis.atlassian.net/browse/PIA-595

File: ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt (hunk 10 above)
Comment (a-szotyori) [reply ginivision/src/main/java/net/gini/android/vision/digitalinvoice/details/LineItemDetailsActivity.kt]: Customization stuff I want to do in this ticket https://ginis.atlassian.net/browse/PIA-595

File: screenapiexample/src/main/java/net/gini/android/vision/screen/ExtractionsActivity.kt (hunk 15 above)
Comment (a-szotyori) [reply screenapiexample/src/main/java/net/gini/android/vision/screen/ExtractionsActivity.kt]: 😄 🙈

File: screenapiexample/src/main/java/net/gini/android/vision/screen/MainActivity.java (hunk 16 above)
Comment (a-szotyori) [reply screenapiexample/src/main/java/net/gini/android/vision/screen/MainActivity.java]: 😄 🙈

**Pull request merged**